Extension/src/cppTools.ts

```diff
@@ -58,22 +58,25 @@ export class CppTools implements CppToolsTestApi {
     public registerCustomConfigurationProvider(provider: CustomConfigurationProvider): void {
         let providers: CustomConfigurationProviderCollection = getCustomConfigProviders();
         if (providers.add(provider, this.version)) {
-            let added: CustomConfigurationProvider1 = providers.get(provider);
-            let settings: CppSettings = new CppSettings();
-            if (settings.loggingLevel === "Information" || settings.loggingLevel === "Debug") {
-                getOutputChannel().appendLine(localize("provider.registered", "Custom configuration provider '{0}' registered", added.name));
+            let addedMayBeUndefined: CustomConfigurationProvider1 | undefined = providers.get(provider);
```

Comment:
if the check above succeeds, shouldn't we be able to assert that this is non-null/undefined? (note the `!` at the end below)

```
let added: CustomConfigurationProvider1 = providers.get(provider)!;
```

Reply:
That would work, if we didn't have `no-non-null-assertion` enabled.  The check here is only necessary due to use of a lambda in a forEach(), which loses its promotion.  It looks like using 'const' instead of 'let' is a way around that issue.